caddy.go

```diff
@@ -877,3 +877,7 @@ type CtxKey string
 // context.WithValue to access the original request URI that accompanied the
 // server request. The associated value will be of type string.
 const URLPathCtxKey CtxKey = "url_path"
+
+// URIxRewriteCtxKey is a context key used to store original unrewritten
+// URI in context.WithValue
+const URIxRewriteCtxKey CtxKey = "caddy_rewrite_original_uri"
```

Comment:
@mholt I seem to get a circular reference between `httpserver` and `staticfiles` since `httpserver`  imports `staticfile` and in order to see the context keys in the `httpserver` package `staticfile` would need to import `httpserver`.  Am I missing something obvious here?